src/modules/auth/auth.controller.ts

```diff
@@ -143,4 +148,54 @@ export class AuthController extends Controller {
       data: { message: "Successfully signedout from all devices" },
     });
   }
+
+  @Post("/forgotPassword")
+  @Middlewares(validateData(authValidations.forgotPass))
+  public async forgotPassword(
+    @Body() body: AuthVerifyEmail
+  ): Promise<APIResponse<Message>> {
+    const user = await userService.fetchByEmail(body.email);
+
+    if (!user) {
+      this.setStatus(statusConst.notFound.code);
+      return toResponse({ error: statusConst.notFound.message });
+    }
+
+    await otpService.send(user, "forgotPassword");
+
+    return toResponse({
+      data: { message: "Forgot password email send successfully" },
+    });
+  }
+
+  @Post("/resetPassword")
+  @Middlewares(authValidations.resetPass)
+  public async resetPassword(
+    @Body() body: AuthResetPass
+  ): Promise<APIResponse<Message>> {
+    const user = await userService.fetchByEmail(body.email);
+    if (!user) {
+      this.setStatus(statusConst.notFound.code);
+      return toResponse({ error: statusConst.notFound.message });
+    }
+
+    const { password, ...restOfBody } = body;
+
+    const verified = await otpService.verify(restOfBody);
+    if (!verified) {
+      this.setStatus(statusConst.unAuthenticated.code);
+      return toResponse({ error: statusConst.unAuthenticated.message });
+    }
+
+    const newPassword = await authSerivce.hashPassword(password);
+
+    await prisma.users.update({
+      where: { id: user.id },
```

Comment:
Once you have removed line 176-180, you can update the user by email instead of id